esprima.js

```diff
@@ -1274,27 +1281,13 @@
 
                     default:
                         if (isOctalDigit(ch)) {
```

Comment:
In fact, I think you can simply switch for '0' and check if the next character is decimal or not.